### ISSUE TITLE
[pd-storage-engine] index out of range

### ISSUE THREAD

**Issue by parisholley** [influxdb/influxdb#4316]
```
goroutine 2056 [running]:
github.com/influxdb/influxdb/tsdb/engine/tsm1.DecodeBlock(0x7fc5b316398e, 0x4, 0xb1974, 0x0, 0x0, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/encoding.go:127 +0x37f
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*cursor).decodeBlock(0xc82cb5e1c0, 0xe003e1982)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/cursor.go:437 +0x152
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*cursor).seekAscending(0xc82cb5e1c0, 0x140a0764ca6e1000, 0x7fc5003e1948, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/cursor.go:288 +0x485
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*cursor).SeekTo(0xc82cb5e1c0, 0x140a0764ca6e1000, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/cursor.go:248 +0x390
github.com/influxdb/influxdb/tsdb/engine/tsm1.(*multiFieldCursor).SeekTo(0xc82cb5e2a0, 0x140a0764ca6e1000, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/engine/tsm1/cursor.go:107 +0xcf
github.com/influxdb/influxdb/tsdb.(*TagsCursor).SeekTo(0xc84a1b2e60, 0x140a0764ca6e1000, 0x0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/cursor.go:365 +0x85
github.com/influxdb/influxdb/tsdb.(*TagSetCursor).Init(0xc825f339e0, 0x140a0764ca6e1000)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/cursor.go:193 +0xbd
github.com/influxdb/influxdb/tsdb.(*AggregateMapper).openMeasurement(0xc83015d680, 0xc82267a2a0, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/mapper.go:479 +0x59d
github.com/influxdb/influxdb/tsdb.(*AggregateMapper).Open(0xc83015d680, 0x0, 0x0)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/mapper.go:433 +0x6de
github.com/influxdb/influxdb/tsdb.(*SelectExecutor).executeAggregate(0xc8419ad3e0, 0xc8458e9740)
	/root/.gvm/pkgsets/go1.5/global/src/github.com/influxdb/influxdb/tsdb/executor.go:372 +0x35c
created by github.com/influxdb/influxdb/tsdb.(*SelectExecutor).Execute

```

**Comment by parisholley** [comment]
this appears to happen when a tag is written without a value. when restarting I get:

```
run: open server: open tsdb store: failed to open shard 2: open engine: unable to parse 'agg_country_fill_rate,country= fill=1 1443979440000000000': missing tag value
```

**Comment by parisholley** [comment]
this may be fixed by #4320, will need to leave server running longer to be sure but so far so good
